interview-questions.md

```diff
@@ -6,62 +6,55 @@ First, without external resources. Challenge yourself to answer from memory as i
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-
 1. What is `this` in JavaScript?
 
-  Your answer:
-
-  Researched answer:
+Your answer: `this` is a keyword that reference to the object when I am inside it.
 
+Researched answer: `this` is an attribute to an object. It is a javascript keyword and can be used to create a variable acessing the value with the object.
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
-
-  Researched answer:
+ Your answer: React is a Javascript library for building user interfaces. It is made up of components, classes. I would use it to have a faster and more powelfull website. 
 
+  Researched answer: React it is a javascript library with an open-source that is well knowing for creating user interfaces more specifically building web applications. I would use it to create a large web application that able to modify data without reloading the page.
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
-
-  Researched answer:
+Your answer: The constructor method.
 
+Researched answer: While searching for the anwser I found out that my previouse awnser was wrong. Render() is actually the only required method in a class component.
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
-
-  Researched answer:
-
+ Your answer: Jsx is a HTML markdown that allows to write in React. HTML is the standard language for documents and JSX is Javascript and HTML together. It is a hybrid language. 
 
+  Researched answer: JSX is an syntax extension to Javascript. It is a language that allow programmers to write XML/HTML language in React. By using JSX the programmer can write succint HTML/XML as structures in the same file that is writing Javascript code. One notable different between HTML and JSX is HTML multiples elements can be returned while in JSX must return one element that wraps all others. 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Yarn is a command use to create a react app.
 
+  Researched answer: Yarn is a JavaScript package manager that creates the file structure of a React app and manages the dependencies. Yarn is a modern package manager for Node.js. It organizes the node_modules. It operates faster thsn other packages, is also more securely and reliably. When you run yarn, The DOM (document object model) that is the visual representation of the code is the one modified when run yarn. 
```

Comment:
Certainly, there is a lot that yarn is doing behind the scenes.  There are a lot more node_modules than we typically care to look at, but yarn manages them for us so as developers we do not need to build all of those smaller building blocks.